controllers/baseboardmanagement_controller.go

```diff
@@ -119,10 +127,12 @@ func (r *BaseboardManagementReconciler) reconcile(ctx context.Context, bm *bmcv1
 	bmcClient, err := r.bmcClientFactory(ctx, bm.Spec.Connection.Host, strconv.Itoa(bm.Spec.Connection.Port), username, password)
 	if err != nil {
 		logger.Error(err, "BMC connection failed", "host", bm.Spec.Connection.Host)
-		result, setConditionErr := r.setCondition(ctx, bm, bmPatch, bmcv1alpha1.ConnectionError, err.Error())
-		if setConditionErr != nil {
-			return result, utilerrors.NewAggregate([]error{fmt.Errorf("failed to set conditions: %v", setConditionErr), err})
+		bm.SetCondition(bmcv1alpha1.Contactable, bmcv1alpha1.BaseboardManagementConditionFalse, bmcv1alpha1.WithBaseboardManagementConditionMessage(err.Error()))
+		result, patchErr := r.patchStatus(ctx, bm, bmPatch)
+		if patchErr != nil {
+			return result, utilerrors.Flatten(utilerrors.NewAggregate([]error{patchErr, err}))
```

Comment:
There's no need to flatten here. You're creating an aggregate from 2 errors and the only way to hit this line is if there are 2 errors so `Flattening` is a noop.